### ISSUE TITLE
Issue running vulkaninfo from MacOS SDK 1.1.82..

### ISSUE THREAD

**Issue by oscarbg** [gfx-rs/portability#147]
EDIT: I see you pointed me libportability_icd was the correct library to use but precompiled library you shared only has libportability.dylib.. so feel free to close if it works with that "_icd" variant..

Hi,
sorry if lame issue and already known but running vulkaninfo using your precompiled Dolphin & RPCS3 binaries I'm getting:

> Using deprecated ICD interface of 'vkGetInstanceProcAddr' instead of 'vk_icdGetInstanceProcAddr' for ICD /lib/libMoltenVK.dylib

and also as you see:
> Segmentation fault: 11
after
> Surface type : VK_MVK_macos_surface

```
==========
VULKANINFO
==========

Vulkan Instance Version: 1.1.82

WARNING: [Loader Message] Code 0 : loader_scanned_icd_add: Using deprecated ICD interface of 'vkGetInstanceProcAddr' instead of 'vk_icdGetInstanceProcAddr' for ICD /lib/libMoltenVK.dylib


Instance Extensions:
====================
Instance Extensions	count = 4
	VK_EXT_debug_report                 : extension revision  9
	VK_EXT_debug_utils                  : extension revision  1
	VK_KHR_surface                      : extension revision 25
	VK_MVK_macos_surface                : extension revision  2
Layers: count = 6
=======
VK_LAYER_GOOGLE_threading (Google Validation Layer) Vulkan version 1.1.82, layer version 1
	Layer Extensions	count = 1
		VK_EXT_debug_report                 : extension revision  6
	Devices 	count = 1
		GPU id       : 0 (AMD Radeon RX Vega 64)
		Layer-Device Extensions	count = 0

VK_LAYER_GOOGLE_unique_objects (Google Validation Layer) Vulkan version 1.1.82, layer version 1
	Layer Extensions	count = 0
	Devices 	count = 1
		GPU id       : 0 (AMD Radeon RX Vega 64)
		Layer-Device Extensions	count = 0

VK_LAYER_LUNARG_core_validation (LunarG Validation Layer) Vulkan version 1.1.82, layer version 1
	Layer Extensions	count = 1
		VK_EXT_debug_report                 : extension revision  6
	Devices 	count = 1
		GPU id       : 0 (AMD Radeon RX Vega 64)
		Layer-Device Extensions	count = 1
			VK_EXT_debug_marker                 : extension revision  4

VK_LAYER_LUNARG_object_tracker (LunarG Validation Layer) Vulkan version 1.1.82, layer version 1
	Layer Extensions	count = 1
		VK_EXT_debug_report                 : extension revision  6
	Devices 	count = 1
		GPU id       : 0 (AMD Radeon RX Vega 64)
		Layer-Device Extensions	count = 1
			VK_EXT_debug_marker                 : extension revision  4

VK_LAYER_LUNARG_parameter_validation (LunarG Validation Layer) Vulkan version 1.1.82, layer version 1
	Layer Extensions	count = 1
		VK_EXT_debug_report                 : extension revision  6
	Devices 	count = 1
		GPU id       : 0 (AMD Radeon RX Vega 64)
		Layer-Device Extensions	count = 1
			VK_EXT_debug_marker                 : extension revision  4

VK_LAYER_LUNARG_standard_validation (LunarG Standard Validation) Vulkan version 1.1.82, layer version 1
	Layer Extensions	count = 1
		VK_EXT_debug_report                 : extension revision  6
	Devices 	count = 1
		GPU id       : 0 (AMD Radeon RX Vega 64)
		Layer-Device Extensions	count = 1
			VK_EXT_debug_marker                 : extension revision  4

Presentable Surfaces:
=====================
GPU id       : 0 (AMD Radeon RX Vega 64)
Surface type : VK_MVK_macos_surface
Segmentation fault: 11
```

**Comment by kvark** [comment]
This is indeed a bug. I tried both the _icd and non-icd versions. The former is apparently not what `vulkaninfo` expects, since it's not looking for `icd` entry points. The non-icd version has an unnecessary panic that occurs, I fixed it in https://github.com/gfx-rs/gfx/pull/2353

Also attaching the output here:  [gfx-macos-info.txt](https://github.com/gfx-rs/portability/files/2328214/gfx-macos-info.txt)
